packages/validator/src/services/block.ts

```diff
@@ -125,19 +125,20 @@ export class BlockProposingService {
         this.validatorStore.getBuilderSelectionParams(pubkeyHex);
       const feeRecipient = this.validatorStore.getFeeRecipient(pubkeyHex);
       const blindedLocal = this.opts.blindedLocal;
+      const useProduceBlockV3 = this.config.getForkSeq(slot) >= ForkSeq.deneb || this.opts.useProduceBlockV3;
```

Comment:
The user provided value would not be evaluated post deneb. I think we want to allow the user to explicitly specify to use v2 apis by passing `useProduceBlockV3=false`

```suggestion
      const useProduceBlockV3 = this.opts.useProduceBlockV3 ?? this.config.getForkSeq(slot) >= ForkSeq.deneb;
```

Reply:
> The user provided value would not be evaluated post deneb. I think we want to allow the user to explicitly specify to use v2 apis by passing useProduceBlockV3=false

yes thats the intend, use provided values are pre deneb

Reply:
Let's clarify with Obol if they still rely on v2 apis post deneb. I think all other clients still support those as well so wouldn't hurt to give users the option to use them.

Reply:
After discussion with Obol, it would be preferred to keep the option to use v2 post deneb to give them some time to rollout complete support for v3.